src/engine/enginemaster.cpp

```diff
@@ -356,6 +359,7 @@ void EngineMaster::processChannels(int iBufferSize) {
         if (m_pEngineEffectsManager) {
             GroupFeatureState features;
             pChannel->collectFeatures(&features);
+            collectMasterSyncFeatures(&features);
```

Comment:
At first I thought this would always overwrite the BPM from `pChannel->collectFeatures` in the line above, but then I saw that `EngineMaster::collectMasterSyncFeatures` checks `!features->has_beat_length_sec`. I think it would help to add a comment here clarifying that the master BPM is only used if `pChannel->collectFeatures` does not report a BPM.

Reply:
Updated.